protos/core.proto

```diff
@@ -487,25 +487,65 @@ message TestResult {
 }
 
 message Field {
+
+  enum Primitive {
+    UNKNOWN = 0;
+    INTEGER = 1;
+    FLOAT = 2;
+    NUMERIC = 5;
+    BOOLEAN = 3;
+    STRING = 4;
+    DATE = 6;
+    DATETIME = 7;
+    TIMESTAMP = 8;
+    TIME = 9;
+    BYTES = 10;
+    // Semi structured data from Snowflake has no strong schema, and could therefore be anything.
+    ANY = 11;
+  }
+
+  enum Flag {
+    UNKNOWN = 0;
+    REPEATED = 1;
+  }
+
   string name = 1;
-  repeated string flags = 4;
+  
+  repeated Flag flag_enums = 6;
```

Comment:
can we make the name change now, to `flags`? (i realise this will make the closed-source merge a bit harder, but seems worth it)